### ISSUE TITLE
Rule Change: support `globalThis` in `no-shadow-restricted-names`

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#19658]
### What rule do you want to change?

`no-shadow-restricted-names`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/*eslint no-shadow-restricted-names: "error"*/

const globalThis = {};

function test(globalThis) {}
```

[**Playground link**](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXM6IFwiZXJyb3JcIiovXG5cbmNvbnN0IGdsb2JhbFRoaXMgPSB7fTtcblxuZnVuY3Rpb24gdGVzdChnbG9iYWxUaGlzKSB7fVxuIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoic2NyaXB0IiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

### What does the rule currently do for this code?

No problem is reported.

### What will the rule do after it's changed?

A problem will be reported for each declaration of `globalThis`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

In ES2020 `globalThis` was added to the [value properties of the global object](https://262.ecma-international.org/11.0/#sec-value-properties-of-the-global-object), but the `no-shadow-restricted-names` rule only reports properties included in [ES5 §15.1.1](https://262.ecma-international.org/5.1/#sec-15.1.1) along with `eval` and `arguments`.

https://github.com/eslint/eslint/blob/48b53d6e79945b4f5f66aa2073c2d51ff7896c7c/docs/src/rules/no-shadow-restricted-names.md?plain=1#L13

This proposal is about adding an option `reportGlobalThis` to the `no-shadow-restricted-names` rule that would report declarations of `globalThis` as well. The new option will be set to `false` by default, and it could be switched to `true` in a major release of ESLint.

**Comment by mdjermanovic** [comment]
As, according to the docs for this rule, one of the criteria for a global variable to be considered a restricted name in this rule is that it is classified as Value Properties of the Global Object in the spec, then it seems like it makes sense to add `globalThis` as well, since it's indeed classified as such. And I agree that this check should be behind the option to avoid a breaking change.

@nzakas what do you think about checking `globalThis` in this rule?

**Comment by fasttime** [comment]
> Should we consider enabling `globalThis` detection by default in v10.0.0?

I'd say yes. That would make the behavior more consistent when backward compatibility isn't needed.

**Comment by fasttime** [comment]
@sethamus You are welcome to submit a pull request. Thanks for asking.
